tests/testthat/test-convert_output_type.R

```diff
@@ -0,0 +1,240 @@
+  sample_outputs <- create_test_sample_outputs()
+  expect_error(
+    convert_output_type(sample_outputs, to = list("cdf" = 0.5)),
+    " cannot be converted to",
+    fixed = TRUE
+  )
+})
+
+test_that("providing incompatible output_type_ids throws an error", {
+  sample_outputs <- create_test_sample_outputs()
+  # mean and median have to NA
+  expect_warning(
+    convert_output_type(sample_outputs, to = list("mean" = 0.5)),
+    "`to` value is incompatible with ",
+    fixed = TRUE
+  )
+  # quantile has numeric to between 0 and 1
+  expect_error(
+    convert_output_type(
+      sample_outputs,
+      to = list("mean" = NA, "quantile" = c(0.25, 1.75))
+    ),
+    "output type IDs should be between 0 and 1.",
+    fixed = TRUE
+  )
+  # quantile has numeric to values
+  expect_error(
+    convert_output_type(
+      sample_outputs,
+      to = list("mean" = NA, "quantile" = c("0.25", 0.75))
+    ),
+    " output type IDs should be numeric",
+    fixed = TRUE
+  )
+
+  # data frame to contains the required columns
+  expect_error(
+    convert_output_type(
+      sample_outputs,
+      to = list("quantile" = data.frame(value = c(0.25, 0.5, 0.75)))
+    ),
+    "`to` did not contain the required column ",
+    fixed = TRUE
+  )
+  # data frame to includes task IDs not present in
+  # the original model output data
+  expect_error(
+    convert_output_type(
+      sample_outputs,
+      to = list("quantile" = expand.grid(group = c(1, 2), output_type_id = c(0.33, 0.66)))
+    ),
+    " element of `to` included ",
+    fixed = TRUE
+  )
+})
+
+test_that("providing output_type_ids with incompatible join columns throws an error", {
+  sample_outputs <- create_test_sample_outputs()
+  # to has more cases of join column values than model_out_tbl (supported but warn)
+  more_levels <- expand.grid(
+    location = c("222", "555", "888"),
+    output_type_id = c(0.25, 0.75),
+    stringsAsFactors = FALSE
+  )
+  expected_outputs <- expand.grid(
+    stringsAsFactors = FALSE,
+    KEEP.OUT.ATTRS = FALSE,
+    model_id = letters[1:4],
+    location = c("222", "888"),
+    horizon = 1, # week
+    target = "inc death",
+    target_date = as.Date("2021-12-25"),
+    output_type = "quantile",
+    value = NA_real_
+  ) |>
+    dplyr::left_join(more_levels, by = "location", relationship = "many-to-many") |>
+    dplyr::arrange(model_id) |>
+    as_model_out_tbl()
+  task_id_cols <- subset_task_id_names(colnames(expected_outputs))
+  expected_outputs$value <- create_test_sample_outputs() |>
+    dplyr::group_by(dplyr::across(dplyr::all_of(c("model_id", task_id_cols)))) |>
+    dplyr::reframe(
+      output_type_id = c(0.25, 0.75),
+      value = stats::quantile(value, probs = c(0.25, 0.75), names = FALSE)
+    ) |>
+    dplyr::pull(value)
+
+  # function call produces correct warning and output
+  expect_warning(
+    actual_outputs <- convert_output_type(sample_outputs, to = list("quantile" = more_levels)),
+    "Some task ID variable combos present in the "
```

Comment:
Should explicitly test for the text produced:

```suggestion
    "Some task ID variable combos present in the `quantile` element of `to` are missing from `model_out_tbl`"
```

Reply:
I've actually been having trouble getting the test to work with the entire message and haven't been able to find a solution yet. Do you have any advice on what to do or resources to look for?

Reply:
Ah yeah, because you used the `arg` keyword, that puts quotes around quantile. I updated my comment to use [raw strings](https://r4ds.hadley.nz/strings#sec-raw-strings)

Reply:
Unfortunately using raw strings doesn't seem to fix the problem. Any other suggestions?

Reply:
> Unfortunately using raw strings doesn't seem to fix the problem. Any other suggestions?

I am confident that they will work. I've tested this on my own machine and it works. Maybe there was a typo or mis-copy?